src/test/java/com/synopsys/integration/alert/web/api/authentication/AuthenticationActionsTestIT.java

```diff
@@ -0,0 +1,249 @@
+ * accordance with the terms of the license agreement you entered into
+ * with Black Duck Software.
+ */
+package com.synopsys.integration.alert.web.api.authentication;
+
+import static junit.framework.TestCase.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+import java.util.List;
+import java.util.Optional;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.lang3.StringUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.mock.web.MockHttpSession;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.dao.DaoAuthenticationProvider;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.ldap.authentication.LdapAuthenticationProvider;
+import org.springframework.security.web.csrf.CsrfTokenRepository;
+
+import com.synopsys.integration.alert.common.action.ActionResponse;
+import com.synopsys.integration.alert.common.descriptor.accessor.AuthorizationUtility;
+import com.synopsys.integration.alert.common.enumeration.AuthenticationType;
+import com.synopsys.integration.alert.common.exception.AlertConfigurationException;
+import com.synopsys.integration.alert.common.exception.AlertDatabaseConstraintException;
+import com.synopsys.integration.alert.common.exception.AlertException;
+import com.synopsys.integration.alert.common.exception.AlertForbiddenOperationException;
+import com.synopsys.integration.alert.common.persistence.model.UserModel;
+import com.synopsys.integration.alert.database.api.DefaultUserAccessor;
+import com.synopsys.integration.alert.mock.model.MockLoginRestModel;
+import com.synopsys.integration.alert.util.AlertIntegrationTest;
+import com.synopsys.integration.alert.util.TestProperties;
+import com.synopsys.integration.alert.util.TestPropertyKey;
+import com.synopsys.integration.alert.util.TestTags;
+import com.synopsys.integration.alert.web.security.authentication.AlertAuthenticationProvider;
+import com.synopsys.integration.alert.web.security.authentication.database.AlertDatabaseAuthenticationPerformer;
+import com.synopsys.integration.alert.web.security.authentication.event.AuthenticationEventManager;
+import com.synopsys.integration.alert.web.security.authentication.ldap.LdapAuthenticationPerformer;
+import com.synopsys.integration.alert.web.security.authentication.ldap.LdapManager;
+
+@Tag(TestTags.CUSTOM_BLACKDUCK_CONNECTION)
+public class AuthenticationActionsTestIT extends AlertIntegrationTest {
```

Comment:
This class is the rename of LoginActionsTestIT.  Tests that were in AuthenticationControllerTestIT have been moved here where appropriate.